### PR TITLE
Upgrade to Ruby 2.6 on Debian

### PR DESCRIPTION
**Pull request by moleskin-smile**
Default Ruby version on Debian is quite old – it is Ruby 2.3.
This version has pretty old version of OpenSSL library that makes
integrating with some outputs problematic.

This change introduces Ruby 2.6 on Debian by utilizing official Ruby
Docker images.

Partially fixes fluent/fluentd-kubernetes-daemonset#241 for me (fluent/fluentd-kubernetes-daemonset#289 is the final fix)

### CODE REVIEW

**Comment by repeatedly** [comment]
Thanks for the patch.
I'm not familiar with ruby base image: https://hub.docker.com/_/ruby
No big differences between ruby and debian?

**Comment by moleskin-smile** [comment]
No big differences. Basically the ruby image depends on debian:stretch-slim just like the fluentd image. It just installs ruby dependencies and compiles it from source. It also defines GEM_HOME and BUNDLE_PATH environment variables and makes $GEM_HOME 777 for user convenience (something I don't like very much, but I can easily trade this for having up to date official debian-based ruby images).

The Dockerfile of image I've used here: https://github.com/docker-library/ruby/blob/995719add69339b78bd8cde46183b4902b761add/2.6/stretch/slim/Dockerfile

Differences in installed packages on amd64:

Additional packages included in ruby-2.6 image:
```
libffi-dev:amd64        3.2.1-6               amd64
libgmp-dev:amd64        2:6.1.2+dfsg-1        amd64
libgmpxx4ldbl:amd64     2:6.1.2+dfsg-1        amd64
libprocps6:amd64        2:3.3.12-3+deb9u1     amd64
libssl-dev:amd64        1.1.0j-1~deb9u1       amd64
libyaml-dev:amd64       0.1.7-2               amd64
procps                  2:3.3.12-3+deb9u1     amd64
zlib1g-dev:amd64        1:1.2.8.dfsg-5        amd64
```

Additional packages included in ruby-2.3 image:

```
rake                    10.5.0-2              all
ruby                    1:2.3.3               amd64
ruby-did-you-mean       1.0.0-2               all
ruby-minitest           5.9.0-1               all
ruby-net-telnet         0.1.1-2               all
ruby-power-assert       0.3.0-1               all
ruby-test-unit          3.1.7-2               all
ruby2.3                 2.3.3-1+deb9u4        amd64
rubygems-integration    1.11                  all
libruby2.3:amd64        2.3.3-1+deb9u4        amd64
libssl1.0.2:amd64       1.0.2r-1~deb9u1       amd64
```

**Comment by repeatedly** [comment]
I see. Thanks for the detailed information.
To change the ruby version, README should be updated together.
https://github.com/fluent/fluentd-docker-image#debian-version
Gem path is changed from 2.3.0 to 2.6.0.

**Comment by moleskin-smile** [comment]
README updated.

**Comment by repeatedly** [comment]
Thanks and sorry for the delay. After back to home, I will recheck and merge.

**Comment by moleskin-smile** [comment]
No problem, please take your time. This is quite big change and should be properly checked. Also this is my first contribution to the project, so it's likely I could have missed something.
Thanks.

File: test/suite.bats [hunk 1]
```diff
@@ -22,14 +22,14 @@
...
-@test "ruby version is 2.3(debian) or 2.5(alpine)" {
+@test "ruby version is 2.6(debian) or 2.5(alpine)" {
   run docker run --rm $IMAGE sh -c "ruby --version | cut -d ' ' -f 2"
   [ "$status" -eq 0 ]
 
   major=$(echo "$output" | cut -d '.' -f 1)
   minor=$(echo "$output" | cut -d '.' -f 2)
   [ "$major" -eq "2" ]
-  [[ "$minor" =~ [3-5]+ ]]
+  [[ "$minor" =~ [3-6]+ ]]
```
Comment (ganmacs) [inline test/suite.bats]: I think `[5-6]` is better since the description of the test is `ruby version is 2.**6**(debian) or 2.**5**(alpine)` .

**Comment by repeatedly** [comment]
I did simple test and it works. I will update major version so new version doesn't break user environment.

**Comment by repeatedly** [comment]
I will publish new version soon.

**Pull request merged**